src/httpserver/http_request.hpp

```diff
@@ -117,27 +117,51 @@ class http_request {
      **/
      const std::map<std::string, std::string, http::header_comparator> get_headers() const;
 
+     /**
+      * Method used to get all header key values passed with the request.
+      * @param keys a vector of string_view into which the keys will be inserted
+     **/
+     void get_header_keys(std::vector<std::string_view>* keys) const;
```

Comment:
Why are we bundling these new public methods into the zero-copy change?

Reply:
The rationale is that you often need to know what keys are present before getting them. The only option otherwise is to the use the "get_headers" which returns everything, also incurring the cost of constructing/allocating the map. We could also change the "get_headers" method and friends to returns maps of string_view as well, but that still incurs the cost of constructing the map. I can add that to the PR if you want. Let me know what you prefer.

Reply:
Having methods to fetch the keys might be fine, but it should be isolated in its own PR not part of this one, so we have the PR doing one job only.